conn.go

```diff
@@ -264,6 +264,15 @@ func (l *Conn) StartTLS(config *tls.Config) error {
 	return nil
 }
 
+// TLSConnectionState returns the current connection's TLS ConnectionState, or an error if not TLS.
+func (l *Conn) TLSConnectionState() (*tls.ConnectionState, error) {
```

Comment:
I'm disinclined to add methods for deep retrieval of fields of private objects.

If anything, a `Conn() net.Conn` function (without changing the interface) would let you get the underlying connection, check cast to tls.Conn yourself, etc.

I also wouldn't change the `Client` interface.